packages/react-day-picker/src/contexts/Focus/FocusContext.tsx

```diff
@@ -59,35 +96,148 @@ export function FocusProvider({
   const focusDayBefore = () => {
     if (!focusedDay) return;
     const before = addDays(focusedDay, -1);
-    setDay(before);
+    changeInitialFocus(before);
     switchMonth(before, numberOfMonths * -1);
   };
   const focusDayAfter = () => {
     if (!focusedDay) return;
     const after = addDays(focusedDay, 1);
-    setDay(after);
+    changeInitialFocus(after);
     switchMonth(after, numberOfMonths);
   };
   const focusWeekBeforeDay = () => {
     if (!focusedDay) return;
     const up = addWeeks(focusedDay, -1);
-    setDay(up);
+    changeInitialFocus(up);
     switchMonth(up, numberOfMonths * -1);
   };
   const focusWeekAfterDay = () => {
     if (!focusedDay) return;
     const down = addWeeks(focusedDay, 1);
-    setDay(down);
+    changeInitialFocus(down);
     switchMonth(down, numberOfMonths);
   };
 
+  const focusFirstDayOfWeek = (): void => {
+    if (!focusedDay) return;
+    const dayToFocus = addDays(
+      startOfWeek(addDays(focusedDay, -1), {
+        weekStartsOn: 0
+      }),
+      1
+    );
+    goToMonth(dayToFocus);
```

Comment:
This should stay the same as the other functions above:

```suggestion
    switchMonth(dayToFocus, numberOfMonths);
```